src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java

```diff
@@ -462,6 +464,69 @@ public ItemStack getSmelting()
         }
     }
 
+    /**
+     *
+     * Event fired before the player is accepted to the world
+     * Cancelling this event will deny the player entry to the server and display the specified reason on their screen
+     *
+     */
+    @Cancelable
+    public static class PrePlayerLoginEvent extends PlayerEvent
+    {
+        private Component denyMessage = new TextComponent("PrePlayerLoginEvent cancelled");
+        private final Connection connection;
+
+        public PrePlayerLoginEvent(Player player, Connection connection)
+        {
+            super(player);
+
+            this.connection = connection;
+        }
+
+        /**
+         *
+         * Cancels the player logging into the server
+         * This will throw a {@link RuntimeException} if you attempt to uncancel a previously cancelled event
+         *
+         * @param cancel If the event should be cancelled (default false)
+         */
+        @Override
+        public void setCanceled(boolean cancel)
+        {
+            if (this.isCanceled() && !cancel)
+            {
+                throw new RuntimeException("You cannot un-cancel the PrePlayerLogin event");
```

Comment:
I think IllegalArgumentException would be more suitable

Reply:
Updated to use IllegalArgumentException